analyses/malawi/notebooks/mwi_ecmwf_monthly_skill_dryspells.md

```diff
@@ -0,0 +1,827 @@
+all_files = glob.glob(os.path.join(ecmwf_country_data_processed_dir, "mwi_seasonal-monthly-single-levels_v5_2*.csv"))
+
+df_from_each_file = (pd.read_csv(f,parse_dates=["date"]) for f in all_files)
+df_for   = pd.concat(df_from_each_file, ignore_index=True)
+```
+
+```python
+#this should be the number of years*number of months
+print(len(all_files))
+```
+
+```python
+#number of years*months till may 2021
+21*12+5
+```
+
+```python
+#for now using mean cell as this requires one variable less to be set (else need to set percentage of cells)
+aggr_meth="mean_cell"
+#for earlier dates, the model included less members --> values for those members are nan --> remove those rows
+df_for = df_for[df_for[aggr_meth].notna()]
+df_for["season_approx"]=np.where(df_for.date.dt.month>=10,df_for.date.dt.year,df_for.date.dt.year-1)
+```
+
+```python
+sel_adm=["Southern"]
+sel_months=[1,2]
+sel_leadtime=[1,2,3,4,5,6]
+seas_years=range(2000,2020)
+
+adm_str="".join([a.lower() for a in sel_adm])
+month_str="".join([calendar.month_abbr[m].lower() for m in sel_months])
+lt_str="".join([str(l) for l in sel_leadtime])
+
+#for this analysis we are only interested in the southern region during the DecJanFeb period, since this is most sensitive to dry spells
+df_for_sel=df_for[(df_for.ADM1_EN.isin(sel_adm))&(df_for.date.dt.month.isin(sel_months))&(df_for.leadtime.isin(sel_leadtime))&(df_for.season_approx.isin(seas_years))]
+```
+
+### Load observational data
+
+```python
+#load the monthly precipitation data
+df_obs_month=pd.read_csv(monthly_precip_path,parse_dates=["date"])
+df_obs_month["date_month"]=df_obs_month.date.dt.to_period("M")
+df_obs_month["season_approx"]=np.where(df_obs_month.date.dt.month>=10,df_obs_month.date.dt.year,df_obs_month.date.dt.year-1)
+
+#select relevant admins and months
+df_obs_month_sel=df_obs_month[(df_obs_month.ADM1_EN.isin(sel_adm))&(df_obs_month.date.dt.month.isin(sel_months))&(df_obs_month.season_approx.isin(seas_years))]
+```
+
+```python
+df_ds=load_dryspell_data(all_dry_spells_list_path)
+```
+
+```python
+#check that same number of months in observed and forecasted
+print("number of months in forecasted data",len(df_for_sel.date.unique()))
+print("number of months in observed data",len(df_obs_month_sel.date.unique()))
+```
+
+### set threshold based on x% probability
+
+```python
+probability=0.5
+```
+
+```python
+#compute the value for which x% of members forecasts the precipitation to be below or equal to that value
+df_for_quant=df_for_sel.groupby(["date","ADM1_EN","leadtime"],as_index=False).quantile(probability)
+df_for_quant["date_month"]=df_for_quant.date.dt.to_period("M")
+```
+
+```python
+df_for_quant[["date","leadtime","mean_cell"]].head()
+```
+
+```python
+#plot the distribution of precipitation. Can clearly see that for leadtime=1 the values are more spread (but leadtime=1 is really the month that is currently occurring so less useful)
```

Comment:
That's interesting and strange, do you know if leadtime=1 is where the initial pertubation happens?

Reply:
yes exactly, so I think that is why